File: csharp/ToolGood.Algorithm/Internals/Functions/Value/Function_ArrayJson.cs
using System;
using System.Collections.Generic;
using System.Text;
using ToolGood.Algorithm.Enums;
using ToolGood.Algorithm.Internals;
using ToolGood.Algorithm.Operands;

namespace ToolGood.Algorithm.Internals.Functions.Value
{
	internal sealed class Function_ArrayJson : Function_N
	{
		public Function_ArrayJson(FunctionBase[] funcs) : base(funcs)
		{
		}

		public override string Name => "ArrayJson";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
		{
			var result = new OperandKeyValueList();
			foreach (var item in funcs) {
				var o = item.Evaluate(engine, tempParameter);
				if(o.IsErrorOrNone) { return o; }
				if(o is OperandKeyValue kv) {
					result.AddValue(kv.Value);
				} else {
					return ParameterError(1);
				}
			}
			return result;
		}
		public override void ToString(StringBuilder stringBuilder, bool addBrackets)
		{
			stringBuilder.Append('{');
			for (int i = 0; i < funcs.Length; i++) {
				if (i > 0) {
					stringBuilder.Append(", ");
				}
				funcs[i].ToString(stringBuilder, false);
			}
			stringBuilder.Append('}');
		}
		public override OperandType GetResultType()
		{
			return OperandType.ARRAYJSON;
		}

		internal override void GetParameterTypes(NoneEngine noneEngine, List<ParameterType> result, OperandType operandType, string op = null, string val = null)
		{
			for(int i = 0; i < funcs.Length; i++) {
				funcs[i].GetParameterTypes(noneEngine, result, OperandType.NONE);
			}
		}
	}

}

File: java/src/main/java/toolgood/algorithm/internals/functions/value/Function_ArrayJson.java
package toolgood.algorithm.internals.functions.value;

import java.util.List;

import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.ParameterType;
import toolgood.algorithm.internals.functions.FunctionBase;
import toolgood.algorithm.internals.functions.Function_N;
import toolgood.algorithm.internals.functions.NoneEngine;
import toolgood.algorithm.operands.OperandKeyValue;
import toolgood.algorithm.operands.OperandKeyValueList;

public final class Function_ArrayJson extends Function_N {
    public Function_ArrayJson(FunctionBase[] funcs) {
        super(funcs);
    }

    @Override
    public String Name() {
        return "ArrayJson";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, java.util.function.BiFunction<AlgorithmEngine, String, Operand> tempParameter) throws Exception {
        OperandKeyValueList result = new OperandKeyValueList();
        for (FunctionBase item : funcs) {
            Operand o = item.Evaluate(engine, tempParameter);
            if (o.IsErrorOrNone()) {
                return o;
            }
            if (o instanceof OperandKeyValue) {
                result.AddValue(((OperandKeyValue) o).Value());
            } else {
                return ParameterError(1);
            }
        }
        return result;
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        stringBuilder.append('{');
        for (int i = 0; i < funcs.length; i++) {
            if (i > 0) {
                stringBuilder.append(", ");
            }
            funcs[i].toString(stringBuilder, false);
        }
        stringBuilder.append('}');
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.ARRAYJSON;
    }

    @Override
    public void GetParameterTypes(NoneEngine noneEngine, List<ParameterType> result, OperandType operandType, String op, String val) {
        for (int i = 0; i < funcs.length; i++) {
            funcs[i].GetParameterTypes(noneEngine, result, OperandType.NONE);
        }
    }
}
